app/controllers/application_controller.rb

```diff
@@ -1331,11 +1331,16 @@ def get_view_process_search_text(view)
                "%#{stxt}%"
              end
 
+      id = @search_text if @search_text =~ /^\d+$/
       return (
-        if ::Settings.server.case_sensitive_name_search
-          ["#{view.db_class.table_name}.#{view.col_order.first} like ? escape '`'", stxt]
+        if id
+          ["#{view.db_class.table_name}.id = ?", id]
```

Comment:
There's an issue here: If I would have a VM with a name that includes a number such as "My_cool_42_vm" and I'd search for the "42" I will not find my VM after this change :-(

Reply:
I'd suggest changing it to be an "OR" search instead of replacing the original behavior.

Does it make sense?

Reply:
ping @Loicavenel

Reply:
Good catch thank @martinpovolny I didn't test this case